tonic/src/status.rs

```diff
@@ -45,6 +44,31 @@ pub struct Status {
     /// If the metadata contains any headers with names reserved either by the gRPC spec
     /// or by `Status` fields above, they will be ignored.
     metadata: MetadataMap,
+    /// Optional underlying `h2::Error` if this `Status` was converted from a `h2::Error`.
+    #[cfg(feature = "transport")]
+    h2_error: Option<h2::Error>,
+}
+
+impl Clone for Status {
+    fn clone(&self) -> Self {
+        Status {
+            code: self.code,
+            message: self.message.clone(),
+            details: self.details.clone(),
+            metadata: self.metadata.clone(),
+            // The following contortion to clone `h2_error` is required because `h2::Error`
+            // cannot implement `Clone` because one of the `Kind` variants  stores a
+            // `std::io::Error` (which does not implement `Clone`). Since we know that
+            // `h2_error` will always have a reason, just extract the reason and create a new
+            // `h2::Error` from it.
```

Comment:
```suggestion
            // The following contortion to clone `h2_error` is required because `h2::Error`
            // cannot implement `Clone` because one of the `Kind` variants  stores a
            // `std::io::Error` (which does not implement `Clone`).
            //
            // We know that if `h2_error` is a `Some` it will have a `reason` since it was
            // created from that `reason`. If its a `None` this clone wont change anything.
```

nit: I find this wording a bit more clearer.

Reply:
Made this change plus fixed some grammatical nits (e.g., `wont`).